prophic3/array_abstractor.cpp

```diff
@@ -452,6 +461,44 @@ void ArrayAbstractor::abstract_array_terms()
   }
 }
 
+void ArrayAbstractor::abstract_large_integer_values() {
+  TermSet integer_vals;
+  detect_integer_values(msat_env_, conc_ts_.init(), integer_vals);
+  detect_integer_values(msat_env_, conc_ts_.trans(), integer_vals);
+  detect_integer_values(msat_env_, conc_ts_.prop(), integer_vals);
+
+  msat_term zero = msat_make_number(msat_env_, "0");
+  msat_term hundred = msat_make_number(msat_env_, "100");
+
+  msat_term t_ = msat_make_true(msat_env_);
+  msat_term gt_100;
+  msat_term lt_0;
+  msat_type inttype = msat_get_integer_type(msat_env_);
+  for (auto iv : integer_vals) {
+    gt_100 = msat_make_not(msat_env_, msat_make_leq(msat_env_, iv, hundred));
+    // TODO use this for very negative numbers
```

Comment:
i think we don't need to do a special handling of negative numbers, because a negative number in msat_term is expressed as (- number). e.g. (- 10)

Reply:
Great! I removed that TODO -- that makes it really clean.